src/network/NetworkKey.cpp

```diff
@@ -0,0 +1,436 @@
+#pragma region Copyright (c) 2014-2016 OpenRCT2 Developers
+/*****************************************************************************
+ * OpenRCT2, an open source clone of Roller Coaster Tycoon 2.
+ *
+ * OpenRCT2 is the work of many authors, a full list can be found in contributors.md
+ * For more information, visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * A full copy of the GNU General Public License can be found in licence.txt
+ *****************************************************************************/
+#pragma endregion
+
+#ifndef DISABLE_NETWORK
+
+#include "NetworkKey.h"
+#include "../diagnostic.h"
+
+#include <openssl/evp.h>
+#include <openssl/rsa.h>
+#include <openssl/pem.h>
+#include <vector>
+
+#define KEY_LENGTH_BITS 2048
+#define KEY_TYPE EVP_PKEY_RSA
```

Comment:
This is just a safeguard, if in future someone wants to change key type to a different algorithm.